core/unit_test/TestReducers.hpp

```diff
@@ -319,6 +329,83 @@ struct TestReducers {
       value = value || values(i);
     }
   };
+
+  // get number of teams for TeamPolicy depending on the tested type
+  constexpr static int get_num_teams() {
+    if constexpr (sizeof(Scalar) == 1) {
+      return 126;
+    } else if constexpr (std::is_same_v<Scalar,
+                                        Kokkos::Experimental::bhalf_t>) {
+      return 256;
+    }
+
+    return 1024;
+  }
+
+  static void test_sum_team_policy(int N, SumFunctor f, Scalar reference_sum) {
+#ifdef KOKKOS_ENABLE_OPENACC
+    if constexpr (std::is_same_v<ExecSpace, Kokkos::Experimental::OpenACC> &&
+                  (std::is_same_v<Scalar, size_t> ||
+                   std::is_same_v<Scalar, double>)) {
+      return;  // FIXME_OPENACC
+    }
+#endif
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+    if constexpr (std::is_same_v<ExecSpace,
+                                 Kokkos::Experimental::OpenMPTarget>) {
+      return;  // FIXME_OPENMPTARGET
+    }
+#endif
+#ifdef KOKKOS_ENABLE_SYCL
+    if constexpr (std::is_same_v<ExecSpace, Kokkos::Experimental::SYCL>) {
+      return;  // FIXME_SYCL
+    }
```

Comment:
What's the problem here? The `SYCL` implementation performs even the copies with `value_type`/`pointer_type` (and doesn't use `word_size_type`) so I wouldn't expect there to be a problem.

Reply:
This is now restricted to skip `reducers_struct` (which uses `array_reduce<float, 1>` etc.) and 
`int8_t` tests.

For `int8_t` we get `CUDA_ERROR_MISALIGNED_ADDRESS` error:
```
5: [ RUN      ] sycl.reducers_int8_t
5: 
5: PI CUDA ERROR:
5: 	Value:           716
5: 	Name:            CUDA_ERROR_MISALIGNED_ADDRESS
5: 	Description:     misaligned address
5: 	Function:        operator()
5: 	Source Location: /root/intel-llvm-mirror/sycl/plugins/cuda/pi_cuda.cpp:2553
5: 
5: 
5: PI CUDA ERROR:
5: 	Value:           716
5: 	Name:            CUDA_ERROR_MISALIGNED_ADDRESS
5: 	Description:     misaligned address
5: 	Function:        cuda_piextUSMFree
5: 	Source Location: /root/intel-llvm-mirror/sycl/plugins/cuda/pi_cuda.cpp:4953
5: 
5: terminate called after throwing an instance of 'sycl::_V1::runtime_error'
5:   what():  Native API failed. Native API returns: -999 (Unknown PI error) -999 (Unknown PI error)
```

`reducers_struct` simply hangs and causes a timeout. I'll be looking closer into those.